### PR TITLE
Add iter function to messages resource

### PR DESCRIPTION
**Pull request by MikeBenza**
Added the `iter` function to the messages resource which allows
the user to specify `DateSent<` and `DateSent>` parameters.

### CODE REVIEW

**Comment by MikeBenza** [comment]
@ihumanable: is there anything I need to do to get this reviewed?

**Comment by MikeBenza** [comment]
@ihumanable @jingming Any feedback on this?

File: twilio/rest/resources/messages.py [hunk 1]
```diff
@@ -142,6 +142,22 @@ def list(self, from_=None, before=None, after=None, date_sent=None, **kw):
         kw["DateSent"] = parse_date(date_sent)
         return self.get_instances(kw)
 
+    @normalize_dates
+    def iter(self, from_=None, to=None, before=None, after=None,
+             date_sent=None, **kwargs):
+        """
+        Returns an iterator of :class:`Call` resources.
```
Comment (ihumanable) [inline twilio/rest/resources/messages.py]: References `Call` class, should reference `Message`

**Comment by efossier** [comment]
+1

**Pull request merged**